### ISSUE TITLE
constexpr constructors

### ISSUE THREAD

**Issue by timblechmann** [boostorg/static_string#27]
there are a few small problems with using `static_string` in constexpr.

one cannot create a static_string from a character literal:
```
constexpr boost::static_string<16> s{"yada"};
```

one cannot return a static_string from a constexpr function:
```
constexpr boost::static_string<16> gen()
{
    boost::static_string<16> ret;
    return ret;
}

constexpr auto x = gen();
```

**Comment by vinniefalco** [comment]
Does that work in C++11 ?

**Comment by alandefreitas** [comment]
This should have been resolved in #32
